### PR TITLE
[50] Add web variant

### LINKED ISSUE #21: Add Procfile management tool for application
## Why
As having more than one process is very common, Procfile is preferred tool to manage more than one process application.

## Acceptance Criteria
- Add [forego](https://github.com/ddollar/forego) or [goreman](https://github.com/mattn/goreman).
 
## Who Benefits?
Developers

### LINKED ISSUE #50: Add web variant
## Why

Sometimes we still need to build a website with Gin framework. So we should add some basic components to create the web faster

## Acceptance Criteria

- Implemented in https://github.com/nimblehq/recruitmate-web/pull/12
- Add variants selection
  - API only
  - Web only
  - Both
- Include unit test

### PR DESCRIPTION
**Pull request by hoangmirs**
https://github.com/nimblehq/git-template/issues/50

## What happened 👀

Add Web variant to the template.
 
## Insight 📝

- [x] Add source code
- [x] Add variants choice variable
    Issue: Cookiecutter does not have conditional variable, so we cannot hide some questions based on the variant selected.
    Ex:
    > If a user select API only variant, we cannot hide the question "Which CSS framework addon?"
     
    Solution retain: 
    - Reduce number of user inputs by asking multiple choice questions instead of y/n questions:
    Ex: Which variant? 1. API only 2. Web only 3. Both
    - Mark explicitely when an addon is not ok for a given variant
    Ex: "Which CSS Framework? (select 'none' if you selected the API only variant) 1. none 2. Bootstrap 3. Tailwind
    - Rewrite the cookiecutter variable names to keep template code maintainable
- [ ] ~Update instructions~ We don't need to add it anymore as we can follow the cookiecutter instructions
- [x] Add post-hooks to remove the redundant files and run `npm install`
- [x] Add tests
 
### Updated template wizard
```
app_name [default-app-name]: blog_web
Select variant:
1 - API only
2 - Web only
3 - Both
Choose from 1, 2, 3 [1]:
```

### Given a Web application
| Config files added | Web folder added |
|---|---|
| image | image |

### Given API only

image
It removes the Web folder as well as the config files:
image

### Test are passing
image

### CODE REVIEW

**Comment by malparty** [comment]
@hoangmirs  shall we add `/node_modules` to the project .gitignore file as well? 

```
{% if cookiecutter._web_variant == "y" %}/node_modules{% endif %}
```

**Comment by gutakk** [comment]
> @hoangmirs shall we add `/node_modules` to the project .gitignore file as well?
> 
> ```
> {% if cookiecutter._web_variant == "y" %}/node_modules{% endif %}
> ```

For me, I think we should.

File: {{cookiecutter.app_name}}/README.md [hunk 1]
```diff
@@ -22,7 +24,9 @@ To start the development server, `.env` file must be created.
 
 - [`air`](https://github.com/cosmtrek/air) is used for live reloading
 
-- [`goose`](https://github.com/pressly/goose) is used for database migration. 
+- [`goose`](https://github.com/pressly/goose) is used for database migration.
+
+{% if cookiecutter._web_variant == "yes" %}- [`forego`](https://github.com/ddollar/forego) manages Procfile-based applications.{% endif %}
```
Comment (malparty) [inline {{cookiecutter.app_name}}/README.md]: @hoangmirs  is there any reason why this line is only for Web variant? 
For what I see in the MakeFile, we use forego also for non-web variant. 
Thanks! :) 

replied = we keep forego for both Web and non-Web projects. So condition removed here.
Solved.

**Review by malparty — APPROVED** [review]

**Review by gutakk — COMMENTED** [review]
@malparty Can you add proof of work here?

**Comment by malparty** [comment]
> @malparty Can you add proof of work here?

@gutakk Done ✅ Let me know if it is OK ;-)

**Review by gutakk — APPROVED** [review]

**Pull request merged**